.gitlab/ci/bucket-upload.yml

```diff
@@ -0,0 +1,147 @@
+        exit 1
+      else
+        echo "User '${GITLAB_USER_NAME}' is allowed to upload packs / force upload packs."
+      fi
+    fi
+
+.install_packs_in_server:
+  needs: ["create-instances"]
+  stage: run-instances
+  artifacts:
+    expire_in: 48 hrs
+    paths:
+      - /builds/xsoar/content/artifacts/*
+  variables:
+    INSTANCE_CREATED: "true"
+    SSH_TUNNEL_TIMEOUT: 10
+    TIME_TO_LIVE: ""
+  extends:
+    - .bucket-upload-rule
+    - .default-job-settings
+  script:
+    - echo "======= Download configuration ========"
+    - ./Tests/scripts/download_demisto_conf.sh  >> $ARTIFACTS_FOLDER/logs/installations.log
+    - *start_tunnel
+    - echo "=====Get Instance Variables====="
+    - echo INSTANCE_ROLE="$INSTANCE_ROLE"
+    - echo INSTANCE_CREATED="$INSTANCE_CREATED"
+    - echo "=====Wait Until Server Ready====="
+    - python3 ./Tests/scripts/wait_until_server_ready.py "$INSTANCE_ROLE"
+    - echo "=====Install Packs====="
+    - ./Tests/Marketplace/install_packs.sh "$INSTANCE_ROLE"
+    - echo "=====Destroy Instances====="
+    - |
+      if [ -n "${CONTRIB_BRANCH}" ]; then
+        echo "Skipping - not running in contribution instance"
+      else
+        python3 ./Tests/scripts/destroy_instances.py $ARTIFACTS_FOLDER $ENV_RESULTS_PATH "$INSTANCE_ROLE" "$TIME_TO_LIVE"
+        export PSWD=$(jq .serverLogsZipPassword < $(cat secret_conf_path) | cut -d \" -f 2)
+        zip -P $PSWD -j $ARTIFACTS_FOLDER/Logs.zip $ARTIFACTS_FOLDER/server*.log || ((($? > 0)) && echo "Didn’t find any server logs, skipping this stage" && exit 0)
+        rm -f $ARTIFACTS_FOLDER/server*.log
+      fi
+
+install-packs-in-server6_0:
+  extends: .install_packs_in_server
+  variables:
+    INSTANCE_ROLE: "Server 6.0"
+
+install-packs-in-server-master:
+  extends: .install_packs_in_server
+  variables:
+    INSTANCE_ROLE: "Server Master"
+
+upload-packs-to-marketplace:
+#  needs: ["run-validations", "run-unittests-and-lint", "install-packs-in-server6_0", "install-packs-in-server-master"]
+  needs: ["install-packs-in-server6_0", "install-packs-in-server-master"]
+  stage: upload-to-marketplace
+  artifacts:
+    expire_in: 48 hrs
+    paths:
+      - /builds/xsoar/content/artifacts/*
+  variables:
+    INSTANCE_ROLE: "Server Master"
+    INSTANCE_CREATED: "true"
+    SSH_TUNNEL_TIMEOUT: 10
+    TIME_TO_LIVE: ""
+  extends:
+    - .bucket-upload-rule
+    - .default-job-settings
+  script:
+    - *start_tunnel
+    - *check_user_permissions_to_upload_packs
+    - echo "=====Upload Packs To Marketplace Storage====="
+    - |
+      if [[ "$CI_COMMIT_BRANCH" == "master" ]] || [[ "$GCS_MARKET_BUCKET" != "marketplace-dist" ]]; then
+        EXTRACT_FOLDER=$(mktemp -d)
+        PACK_ARTIFACTS=$ARTIFACTS_FOLDER/content_packs.zip
+        PACKS_DEPENDENCIES=$ARTIFACTS_FOLDER/packs_dependencies.json
+        CI_COMMIT_BRANCH=${CI_COMMIT_BRANCH:-unknown}
+        GCS_BUILD_BUCKET="marketplace-ci-build"
```

Comment:
I thought we want to upload to a different bucket (until CircleCI build is deprecated) so that we don't overwrite the uploads from the circle build that is running in parallel. Is that not the case?

Reply:
we can just trigger the build and override this but I changed it just in case, good call.